### ISSUE TITLE
rustup tries to update itself and fails with error: Read-only file system

### ISSUE THREAD

**Issue by ivan** [NixOS/nixpkgs#91191]
**Describe the bug**

On nixpkgs master, `rustup update` tries to do a self-update and fails with

```
error: Read-only file system (os error 30) at path "/nix/store/gxx9rxzvmk4mhzamv679gc65njgc754r-rustup-1.21.1/bin/updtest9a6vzy"
```

I believe this is a regression from 20.03, where I never saw this.

**To Reproduce**
1. On nixpkgs master, `nix-env -iA rustup && rustup update`
2. Observe the above-mentioned error

**Additional context**

```
# rustup update
info: syncing channel updates for 'stable-x86_64-unknown-linux-gnu'
info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu'
info: latest update on 2020-06-20, rust version 1.46.0-nightly (2d8bd9b74 2020-06-19)
info: downloading component 'rust-std' for 'wasm32-unknown-unknown'
  9.5 MiB /   9.5 MiB (100 %)   2.6 MiB/s in  3s ETA:  0s
info: downloading component 'rust-src'
info: downloading component 'cargo'
  5.0 MiB /   5.0 MiB (100 %)   2.7 MiB/s in  1s ETA:  0s
info: downloading component 'clippy'
info: downloading component 'rust-docs'
 12.2 MiB /  12.2 MiB (100 %)   2.5 MiB/s in  4s ETA:  0s
info: downloading component 'rust-std'
 16.0 MiB /  16.0 MiB (100 %)   2.7 MiB/s in  6s ETA:  0s
info: downloading component 'rustc'
 47.7 MiB /  47.7 MiB (100 %)   2.6 MiB/s in 18s ETA:  0s
info: downloading component 'rustc-dev'
 70.9 MiB /  70.9 MiB (100 %)   2.5 MiB/s in 28s ETA:  0s
info: removing previous version of component 'rust-std' for 'wasm32-unknown-unknown'
info: removing previous version of component 'rust-src'
info: removing previous version of component 'cargo'
info: removing previous version of component 'clippy'
info: removing previous version of component 'rust-docs'
info: removing previous version of component 'rust-std'
info: removing previous version of component 'rustc'
info: removing previous version of component 'rustc-dev'
info: installing component 'rust-std' for 'wasm32-unknown-unknown'
info: installing component 'rust-src'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'rust-docs'
 12.2 MiB /  12.2 MiB (100 %)   4.3 MiB/s in  3s ETA:  0s
info: installing component 'rust-std'
 16.0 MiB /  16.0 MiB (100 %)   9.4 MiB/s in  1s ETA:  0s
info: installing component 'rustc'
 47.7 MiB /  47.7 MiB (100 %)   9.5 MiB/s in  5s ETA:  0s
info: installing component 'rustc-dev'
 70.9 MiB /  70.9 MiB (100 %)   8.9 MiB/s in  7s ETA:  0s
error: Read-only file system (os error 30) at path "/nix/store/gxx9rxzvmk4mhzamv679gc65njgc754r-rustup-1.21.1/bin/updtest9a6vzy"

# which rustup
/run/current-system/sw/bin/rustup

# rustup --version
rustup 1.21.1 (1970-01-01)

# rustup update
info: syncing channel updates for 'stable-x86_64-unknown-linux-gnu'
info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu'
error: Read-only file system (os error 30) at path "/nix/store/gxx9rxzvmk4mhzamv679gc65njgc754r-rustup-1.21.1/bin/updtesto9M7mI"

# rustup update --no-self-update
info: syncing channel updates for 'stable-x86_64-unknown-linux-gnu'
info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu'

   stable-x86_64-unknown-linux-gnu unchanged - rustc 1.44.1 (c7087fe00 2020-06-17)
  nightly-x86_64-unknown-linux-gnu unchanged - rustc 1.46.0-nightly (2d8bd9b74 2020-06-19)

info: cleaning up downloads & tmp directories
```

**Notify maintainers**
@Mic92 

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.47-hardened, NixOS, 20.09.git.622cec1e30d (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - nixpkgs: `/var/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by Mic92** [comment]
I would say rustup should be patched to default to `--no-self-update` internally.

**Comment by Flakebi** [comment]
Not sure what the problem here is, we build rustup with `--features no-self-update` (I also tried `--features=no-self-update` with `=` instead of space).
Still, I get the same error.

**Comment by Mic92** [comment]
Can you also report this issue upstream?

**Comment by Flakebi** [comment]
Found it, it’s a packaging error, `cargo test` is run without features and overwrites the previously compiled binaries.
I made a fix in #91689.
